### PR TITLE
dialects: Add a stencil.AccesOp -> memref.Load lowering.

### PR DESCRIPTION
**Pull request by PapyChacal**

### CODE REVIEW

File: xdsl/transforms/experimental/ConvertStencilToLLMLIR.py [hunk 1]
```diff
@@ -119,6 +119,49 @@ def match_and_rewrite(self, op: ApplyOp, rewriter: PatternRewriter, /):
...
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: AccessOp, rewriter: PatternRewriter, /):
+        if not isinstance(op.temp.owner, memref.Cast):
+            warn(
+                "stencil.load should have been lowered before lowering related "
+                "stencil.access")
+            return
+
+        if (block := op.parent_block()) is None:
```
Comment (webmiche) [inline xdsl/transforms/experimental/ConvertStencilToLLMLIR.py:132]: Can this actually happen?

**Review by webmiche — APPROVED** [review]
Some mionr comments

File: tests/filecheck/dialects/stencil/test_apply_lowering.mlir [hunk 2]
```diff
@@ -15,7 +15,7 @@
 // CHECK-NEXT: "builtin.module"() ({
 // CHECK-NEXT:     "func.func"() ({
 // CHECK-NEXT:     ^0(%0 : memref<?x?x?xf64>):
-// CHECK-NEXT:         %1 = "memref.cast"(%0) {"stencil_offset" = [4 : i64, 4 : i64, 4 : i64]} : (memref<?x?x?xf64>) -> memref<72x72x72xf64>
+// CHECK-NEXT:         %1 = "memref.cast"(%0) {"stencil_offset" = #stencil.index<[4 : i64, 4 : i64, 4 : i64]>} : (memref<?x?x?xf64>) -> memref<72x72x72xf64>
```
Comment (georgebisbas) [inline tests/filecheck/dialects/stencil/test_apply_lowering.mlir:18]: Why index and not offset?

File: xdsl/transforms/experimental/ConvertStencilToLLMLIR.py (hunk 1 above)
Comment (georgebisbas) [inline xdsl/transforms/experimental/ConvertStencilToLLMLIR.py:132]: Can you add a dosctringon this condition?

Comment (PapyChacal) [reply xdsl/transforms/experimental/ConvertStencilToLLMLIR.py:132]: Nope, it's for type-checking purpose only. Would it be clearer with an assert?

Comment (PapyChacal) [reply xdsl/transforms/experimental/ConvertStencilToLLMLIR.py:132]: Answered on webmiche's comment of the same line

Comment (webmiche) [reply xdsl/transforms/experimental/ConvertStencilToLLMLIR.py:132]: Yep, I think an assert would make more sense here.

File: xdsl/transforms/experimental/ConvertStencilToLLMLIR.py [hunk 3]
```diff
@@ -119,6 +119,49 @@ def match_and_rewrite(self, op: ApplyOp, rewriter: PatternRewriter, /):
...
+            return
+
+        if not isinstance(op.temp.owner.attributes["stencil_offset"],
```
[the remark this answers is not in the document]
Comment (PapyChacal) [reply xdsl/transforms/experimental/ConvertStencilToLLMLIR.py:135]: Assuming the question is why `attributes['stencil_offset']`vs `.stencil_offset`:
It is not a registered attribute of the operation, as it only carries temporary meaning during the lowering and is intended to be removed after it.
Was it your question?

File: xdsl/transforms/experimental/ConvertStencilToLLMLIR.py (hunk 1 above)
Comment (PapyChacal) [reply xdsl/transforms/experimental/ConvertStencilToLLMLIR.py:132]: noted, will do

File: xdsl/transforms/experimental/ConvertStencilToLLMLIR.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (webmiche) [reply xdsl/transforms/experimental/ConvertStencilToLLMLIR.py:135]: Yep, thanks 👍

**Pull request merged**